statement.go

```diff
@@ -7,39 +7,44 @@ import (
 	"database/sql/driver"
 )
 
-type snowflakeStmt struct {
+type SnowflakeStmt struct {
```

Comment:
why do we need to export this? I would've thought that there would be some exported `interface` which clients of this library should be using

Reply:
The `DescribeContext` function isn't in the golang sql package, so we need to refine up to this specific type.

Reply:
ah, that might be good to note here then